### ISSUE TITLE
how webhook works

### ISSUE THREAD

**Issue by PeraDev7** [wallee-payment/php-sdk#15]
hi all, reading the documentation I can't understand how the webhook works. Once the payment process has been done, I need to redirect the user to a page, but from here on it's not clear to me how I can take the data and extrapolate it. I have already set up the webhook link from the panel. thank you

**Comment by theodorspringwallee** [comment]
Dear [PeraDev7](https://github.com/PeraDev7)

If you just want the existing data of a transaction on the success page, I recommend you to pass the transaction ID to the success page and read from there the transaction:
```
// e.g. Success URL: www.your-shop.com/order/successful?transaction_id=5
$transactionId = htmlspecialchars($_GET['transaction_id']);
$transaction = $apiClient->getTransactionService()->read($spaceId, $transaction->getId());
// with e.g. $transaction->getBillingAddress()->getFamilyName(); you can get then the data out of it
```
But it can take longer till the transition between authorize to fulfill is done (for example in deferred mode for credit cards). 
This information will be transmitted via Webhooks to your shop system. 

The URL you define for your Webhooks needs to have a script behind, that is fetching each incoming request (Webhook) from the wallee portal.

Workaround to get transaction updates by Webhooks:
1. First, the client uses a URL with the destination address where he wants to receive the webhooks (to test you can use [Webhook.site](https://webhook.site/)): https://app-wallee.com/s/your-space/webhook/url/list
![image](https://user-images.githubusercontent.com/85299706/203942225-774fc658-8948-4400-8d6c-3169acdb98fc.png)

2. Then the client creates a listener with the type "Transaction" and with the statuses at which the webhooks should be sent: https://app-wallee.com/s/your-space/webhook/listener/list
![image](https://user-images.githubusercontent.com/85299706/203942312-f020a807-1af2-49b2-9acf-26b9ae71ec2b.png)

3. At the defined target URL of the customer, the customer can intercept the webhooks. There is the entity ID (transaction ID included).
4. To get the transaction status and other transaction information, the customer must call the following method with the entity and space ID: https://app-wallee.com/api/client/transaction-service--read

Does this answer your questions?

Best regards

**Comment by PeraDev7** [comment]
Thank you
I wanted to opt for inserting the transaction ID in the success url, but from what I see it is not possible, because the ID is accessible only after creating the transaction, right?

```php
$transactionPayload->setSuccessUrl($urlSuccess);
$transactionPayload->setLineItems(array($lineItem));
$transactionPayload->setAutoConfirmationEnabled(true);
$transaction = $client->getTransactionService()->create($spaceId, $transactionPayload);
$redirectionUrl = $client->getTransactionPaymentPageService()->paymentPageUrl($spaceId, $transaction->getId());
```
Am i wrong? I'm stucked here. How can i get the ID before? Thanks

**Comment by theodorspringwallee** [comment]
Dear [PeraDev7](https://github.com/PeraDev7)

You can do it this way with TransactionPending as a step after the create() and before calling the payment page url:
```
$transaction = $this->client->getTransactionService()->create($this->spaceId, $transactionPayload);

  $transactionPending = new \Wallee\Sdk\Model\TransactionPending([
      "id" => $transaction->getId(), // id of created transaction
      "version" => 1
  ]);

$transactionPending->setSuccessUrl("https://yourshop.com/order/success.php?transaction_id="
. $transaction->getId());
$transactionPending->setFailedUrl("https://yourshop.com/order/failed.php?transaction_id="
. $transaction->getId());

$this->client->getTransactionService()->update($this->spaceId, $transactionPending); // update created transaction

$redirectionUrl = 
$this->client->getTransactionPaymentPageService()->paymentPageUrl($this->spaceId, $transaction->getId());

```
Best regards

**Comment by PeraDev7** [comment]
Thank you!
